### PR TITLE
chore: reclassify common sdk errors

### PR DESCRIPTION
**Pull request by goldbez**
#### Description of changes
This PR adds try-catch blocks to SDK calls in push-resources.ts. These try-catch bocks call a function that checks if the thrown error matches a set of errors that we have pre-determined to be common user errors, and reclassifies them as such. In future PRs, this function can be used when adding try-catch blocks to SDK calls in other parts of the code.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-core/src/errors/amplify-exception.ts [hunk 1]
```diff
@@ -158,6 +158,7 @@ export type AmplifyErrorType =
   | 'ProjectNotFoundError'
   | 'ProjectNotInitializedError'
   | 'PushResourcesError'
+  | 'RateError'
```
Comment (Amplifiyer) [inline packages/amplify-cli-core/src/errors/amplify-exception.ts:161]:
```suggestion
  | 'APIRateExceededError'
```

**Review by Amplifiyer — COMMENTED** [review]
LGTM

**Review by jhockett — COMMENTED** [review]
There's a lot of AWS-SDK calls that won't be checked with this approach. Is there a reason we can't do something in the exception handling logic itself rather than duplicate this call everywhere?

**Comment by goldbez** [comment]
> There's a lot of AWS-SDK calls that won't be checked with this approach. Is there a reason we can't do something in the exception handling logic itself rather than duplicate this call everywhere?

The reason for not doing the error handling in a central location is that the stack track would be destroyed. This PR is only meant to cover the SDK calls in push-resources.ts. In later PRs, I will cover SDK calls in other places.

File: packages/amplify-provider-awscloudformation/src/__tests__/handle-common-sdk-errors.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      try {
+        throw new Error(
+          'User: ARN is not authorized to perform: ACTION:ACTION on resource: RESOURCE because no identity-based policy allows the ACTION:ACTION action',
+        );
+      } catch (error) {
+        throw handleCommonSdkError(error);
+      }
+    };
+
+    confirmCorrectError(toThrowError, 'Update the permissions.');
```
Comment (Amplifiyer) [inline packages/amplify-provider-awscloudformation/src/__tests__/handle-common-sdk-errors.test.ts:35]:
```suggestion
    confirmErrorResolution(toThrowError, 'Update the permissions.');
```

**Review by Amplifiyer — APPROVED** [review]
nit: update the function name otherwise looks good.

**Review by jhockett — APPROVED** [review]

**Pull request merged**